homeassistant/components/freebox/binary_sensor.py

```diff
@@ -44,10 +47,105 @@ async def async_setup_entry(
+        elif node["category"] == FreeboxHomeCategory.DWS:
+            binary_entities.append(FreeboxDwsSensor(hass, router, node))
+
+        for endpoint in node["show_endpoints"]:
+            if (
+                endpoint["name"] == "cover"
+                and endpoint["ep_type"] == "signal"
+                and endpoint.get("value") is not None
+            ):
+                binary_entities.append(FreeboxCoverSensor(hass, router, node))
+
     if binary_entities:
         async_add_entities(binary_entities, True)
 
 
+class FreeboxBinarySensor(FreeboxHomeEntity, BinarySensorEntity):
+    """Representation of a Freebox binary sensor."""
+
+    def __init__(
+        self, hass: HomeAssistant, router: FreeboxRouter, node: dict[str, Any]
+    ) -> None:
+        """Initialize a Freebox binary sensor."""
+        super().__init__(hass, router, node)
+        self._command_trigger = self.get_command_id(
+            node["type"]["endpoints"], "signal", "trigger"
+        )
+
+        self._detection = False
+
+    async def async_update_signal(self):
+        """Watch states."""
+        detection = await self.get_home_endpoint_value(self._command_trigger)
+        if detection is not None:
+            if self._detection == detection:
+                self._detection = not detection
+                self.async_write_ha_state()
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if the binary sensor is on."""
+        return self._detection
+
+
+class FreeboxPirSensor(FreeboxBinarySensor):
+    """Representation of a Freebox motion binary sensor."""
+
+    @property
+    def device_class(self) -> BinarySensorDeviceClass:
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return BinarySensorDeviceClass.MOTION
+
+
+class FreeboxDwsSensor(FreeboxBinarySensor):
+    """Representation of a Freebox door opener binary sensor."""
+
+    @property
+    def device_class(self) -> BinarySensorDeviceClass:
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return BinarySensorDeviceClass.DOOR
+
+
+class FreeboxCoverSensor(FreeboxHomeEntity, BinarySensorEntity):
+    """Representation of a cover Freebox plastic removal cover binary sensor (for some sensors: motion detector, door opener detector...)."""
+
+    def __init__(
+        self, hass: HomeAssistant, router: FreeboxRouter, node: dict[str, Any]
+    ) -> None:
+        """Initialize a cover for another device."""
+        cover_node = next(
+            filter(
+                lambda x: (x["name"] == "cover" and x["ep_type"] == "signal"),
+                node["type"]["endpoints"],
+            ),
+            None,
+        )
+        super().__init__(hass, router, node, cover_node)
+        self._command_cover = self.get_command_id(
+            node["type"]["endpoints"], "signal", "cover"
+        )
+        self._open = self.get_value("signal", "cover")
```

Comment:
Shouldn't be necessary

or call `async_update_node`

Reply:
Done